utils/aws/sqs/main.tf

```diff
@@ -0,0 +1,231 @@
+################################################################################
+# Queue
+################################################################################
+
+locals {
+  name = try(trimsuffix(var.name, ".fifo"), "")
+}
+
+resource "aws_sqs_queue" "this" {
+  count = var.create ? 1 : 0
+
+  content_based_deduplication       = var.content_based_deduplication
+  deduplication_scope               = var.deduplication_scope
+  delay_seconds                     = var.delay_seconds
+  fifo_queue                        = var.fifo_queue
+  fifo_throughput_limit             = var.fifo_throughput_limit
+  kms_data_key_reuse_period_seconds = var.kms_data_key_reuse_period_seconds
+  kms_master_key_id                 = var.kms_master_key_id
+  max_message_size                  = var.max_message_size
+  message_retention_seconds         = var.message_retention_seconds
+  name                              = var.use_name_prefix ? null : (var.fifo_queue ? "${local.name}.fifo" : local.name)
+  name_prefix                       = var.use_name_prefix ? "${local.name}-" : null
```

Comment:
I think you should just use name prefix no matter what. Why give the choice to the user?

Reply:
Done !